src/main/java/cz/centrum/haffner/SimpleTrainingDavid/AppServices/MetricsInfoService.java

```diff
@@ -10,167 +13,26 @@
 import java.util.HashMap;
 import java.util.Map;
 
-// algorithm for returning of the proper MetricsInfo set
+// algorithm for returning of the proper MetricsInfo data set
+@Service
 public class MetricsInfoService {
+    @Autowired
+    private MetricsInfoDataSet metricsInfoDataSet;
 
-    // TODO: net approach doesn´t work
-    // private static final String FILE_PATH = "https://github.com/DavidHaffner/SimpleTrainingDavid/tree/dev/logs/MCP_20180131.json";
-    private static final String FILE_PATH_PREFIX = "C://work/SimpleTrainingDavid/src/main/resources/MCPData/";
-
-    private int missingFieldsRowsCounter =0;        // Number of rows with missing fields
-    private int blankContentMessagesCounter =0;     // Number of messages with blank content
-    private int fieldsErrorsRowsCounter =0;        // Number of rows with fields errors
-    private int groupedCallsCounter =0;            // Number of calls TODO: origin/destination grouped by country code
-    private float koToOkRatio =0.0f;               // Relationship between OK/KO calls
-    private float averageCallDuration =0.0f;            // Average call duration TODO: grouped by country code
-    private String givenWordsRanking ="UNDER CONSTRUCTION YET";  // TODO: Word occurrence ranking for the given words in message_content field
+    @Autowired
+    private OneDayFileJsonParser oneDayFileJsonsParser;
 
 
     public MetricsInfoService(){
```

Comment:
Java automaticky vytváří defaultní konstruktor (bezparametrický) pro objekty, takže tento kód v budoucnu nemusíš přidávat - nedělá se to.
Konstruktor se explicitně vytváří, pokud v něm chceš něco inicializovat po svém z případných parametrů do něj předaných ;-)

Reply:
ok